user-interface/src/lib/hooks/UseApi.ts

```diff
@@ -42,3 +42,17 @@ export interface ApiClient {
   put(path: string, body: object, options?: ObjectKeyVal): Promise<ResponseData>;
   getQueryStringsToPassthrough(search: string, options: ObjectKeyVal): ObjectKeyVal;
 }
+
+export interface GenericApiClient {
```

Comment:
This looks odd to me. Not sure if I understand the responsibility of this GenericApiClient and its implementation compared to the ApiClient.

Reply:
Add some notes about the intention of the `GenericApiClient`.